modules/backendscheduler/backendscheduler.go

```diff
@@ -59,20 +64,47 @@ func New(cfg Config, store storage.Store, overrides overrides.Interface, reader
 	}
 
 	s := &BackendScheduler{
-		cfg:         cfg,
-		store:       store,
-		overrides:   overrides,
-		work:        work.New(cfg.Work),
-		reader:      reader,
-		writer:      writer,
-		curPriority: tenantselector.NewPriorityQueue(),
+		cfg:        cfg,
+		store:      store,
+		overrides:  overrides,
+		work:       work.New(cfg.Work),
+		reader:     reader,
+		writer:     writer,
+		mergedJobs: make(chan *work.Job, 100),
```

Comment:
would prefer an unbuffered channel here or at least very lightly buffered.  we shouldn't have any starvation issues on the workers b/c of how quick we can generate jobs and completely unbuffered would make us the most responsive to changes. the provider channels could be unbuffered as well as this would make a very easy to understand "machine" that delivered jobs from the providers to the Next() function